File: 1513-number-of-substrings-with-only-1s/1513-number-of-substrings-with-only-1s.cs
public class Solution {
    public int NumSub(string s) {
        var MOD = 1_000_000_007;
        long ans = 0, ones = 0;

        foreach(var ch in s){
            if(ch == '1')
                ones++;
            else if (ones != 0) {
                ans += (ones + 1) * ones / 2;
                ans %= MOD;
                ones = 0;
            }
        }

        if(ones != 0){
            ans += (ones + 1) * ones / 2;
            ans %= MOD;
        }

        return (int)ans;
    }
}

File: 1513-number-of-substrings-with-only-1s/1513-number-of-substrings-with-only-1s.py
class Solution:
    def numSub(self, s: str) -> int:
        MOD = 1_000_000_007
        ans = ones = 0

        for ch in s:
            if ch == '1':
                ones += 1
            elif ones != 0:
                ans += (ones + 1) * ones // 2
                ans %= MOD
                ones = 0

        if ones != 0:
            ans += (ones + 1) * ones // 2
            ans %= MOD

        return ans